UNet functional and performance demo crash

### Summary
The UNet demo script does currently crashes when being run.

Reproduce by following the instructions in the README:

```sh
# Run functional demo
pytest --disable-warnings models/experimental/functional_unet/tests/test_unet_shallow_functional.py

# Run perf demo
pytest --disable-warnings models/experimental/functional_unet/tests/test_unet_shallow_performance.py::test_unet_model_performance
```

These both fail with the following error:

```sh
 RuntimeError: TT_THROW @ ../ttnn/cpp/ttnn/operations/matmul/matmul.cpp:66: tt::exception
E       info:
E       ttnn.matmul: The width of the first tensor must be equal to the height of the second tensor
```

This model does not appear to be covered under CI, so it has been broken for some unknown amount of time.

### Details

- Crash occurs on final layer (`self.output_layer`), where a 1x1 convolution gets converted to a matmul.
- We are trying to multiply `(337920, 16[32]) x (32, 32)`.

Here is test case that uses the correct shapes:

```python
import ttnn
import torch


def test_unet_1x1_conv(device):
    core_grid = ttnn.CoreGrid(x=8, y=8)
    x = ttnn.from_torch(
        torch.rand((1, 1, 337920, 16)), device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.L1_MEMORY_CONFIG
    )
    print(x.shape)

    sharding_x = ttnn.create_sharded_memory_config((1, 1, 337920, 32), core_grid, ttnn.ShardStrategy.HEIGHT)
    x = ttnn.to_memory_config(x, sharding_x)
    y = ttnn.from_torch(
        torch.rand((1, 1, 32, 32)), device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG
    )
    print(x.shape, y.shape)

    z = ttnn.linear(x, y)
```

--- a/CODEOWNERS
+++ b/CODEOWNERS
@@ -140,6 +140,7 @@ models/demos/tg/falcon7b @skhorasganiTT @djordje-tt @uaydonat
 models/demos/grayskull @uaydonat @eyonland
 models/demos/resnet @mywoodstock @shwetankTT @tt-aho
 models/demos/ttnn_resnet @mywoodstock @shwetankTT @tt-aho
+models/experimental/functional_unet @esmalTT @uaydonat @mywoodstock
 models/perf/ @uaydonat @tt-rkim
 models/perf/benchmarking_utils.py @skhorasganiTT @tt-rkim
 
@@ -158,6 +159,7 @@ tests/python_api_testing/conv/ @mywoodstock @sankarmanoj-tt
 tests/python_api_testing/unit_testing/fallback_ops @tt-aho
 tests/ttnn/integration_tests/stable_diffusion @esmalTT @uaydonat @mywoodstock
 tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py @esmalTT @uaydonat @mywoodstock
+tests/ttnn/integration_tests/unet @esmalTT @uaydonat @mywoodstock
 scripts/profiler/ @mo-tenstorrent
 scripts/docker @ttmchiou @TT-billteng @tt-rkim
 

--- a/models/experimental/functional_unet/tests/test_unet_shallow_functional.py
+++ b/models/experimental/functional_unet/tests/test_unet_shallow_functional.py
@@ -61,8 +61,5 @@ def test_unet_pcc(device, perf_mode, batch, groups):
         output_tensor = torch.permute(output_tensor, (0, 3, 1, 2))
         output_tensor = output_tensor.to(torch_input_tensor.dtype)
 
-        # Disable pcc checking due to hang
-        # if device.arch() == ttl.device.Arch.WORMHOLE_B0:
-        #    assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.99)
-        # else:
-        #    assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.97)
+        assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.99)
+

--- a/models/experimental/functional_unet/tt/unet_shallow_ttnn.py
+++ b/models/experimental/functional_unet/tt/unet_shallow_ttnn.py
@@ -108,6 +108,7 @@ class UNet:
         self.c8_2 = parameters.c8_2
         self.c8_3 = parameters.c8_3
         self.output_layer = parameters.output_layer
+        self.cache = {}
 
     def __call__(self, device, input_tensor, original_shape, perf_mode=False):
         nhw = original_shape[-4] * original_shape[-2] * original_shape[-1]
@@ -251,5 +252,20 @@ class UNet:
         output_tensor = self.c8(output_tensor)
         output_tensor = self.c8_2(output_tensor)
         output_tensor = self.c8_3(output_tensor)
-        output_tensor = self.output_layer(output_tensor)
+        output_tensor, _, _, self.output_layer.conv.weight, self.output_layer.conv.bias = ttnn.conv2d(
+            input_tensor=output_tensor,
+            weight_tensor=self.output_layer.conv.weight,
+            bias_tensor=self.output_layer.conv.bias,
+            in_channels=self.output_layer.in_channels,
+            out_channels=self.output_layer.out_channels,
+            input_height=self.output_layer.input_height,
+            input_width=self.output_layer.input_width,
+            batch_size=self.output_layer.batch_size,
+            kernel_size=(1, 1),
+            stride=(1, 1),
+            padding=(0, 0),
+            conv_op_cache=self.cache,
+            device=device,
+        )
+
         return ttnn.from_device(output_tensor)

